### PR TITLE
[TF] Prevent concurrent istioctl invoke

### PR DESCRIPTION
**Pull request by nmittler**
**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/test/framework/components/istioctl/kube.go [hunk 1]
```diff
@@ -26,6 +27,12 @@ import (
 	"istio.io/istio/pkg/test/scopes"
 )
 
+var (
+	// We cannot invoke the istioctl library concurrently due to the number of global variables
+	// https://github.com/istio/istio/issues/37324
+	invokeMutex sync.Mutex
```
Comment (howardjohn) [inline pkg/test/framework/components/istioctl/kube.go:33]: at one point I thought we installed multi clusters in parallel. do we not do that already, or does this PR stop making that happen?

**Review by howardjohn — COMMENTED** [review]
lgtm if it works. Was there issues?

**Comment by nmittler** [comment]
@howardjohn I'm doing this purely as a reaction to realizing that this is fundamentally broken. If we are currently doing things in parallel, I suspect it's an issue waiting to happen.

@stevenctl any concerns with this?

**Comment by stevenctl** [comment]
+1, There is a lot of global use in istioctl.

File: pkg/test/framework/components/istioctl/kube.go (hunk 1 above)
Comment (stevenctl) [reply pkg/test/framework/components/istioctl/kube.go]: Good point..

Comment (nmittler) [reply pkg/test/framework/components/istioctl/kube.go]: @stevenctl  How is our installation safe if we're invoking istioctl concurrently?

**Comment by nmittler** [comment]
@stevenctl It looks like `integ-pilot-multicluster_istio` takes about 5 min longer with this PR. If that's a concern, let's discuss possible work-arounds.

File: pkg/test/framework/components/istioctl/kube.go (hunk 1 above)
Comment (stevenctl) [reply pkg/test/framework/components/istioctl/kube.go]: It's not.. but this part also doesn't seem to be horribly affected. 

We could also workaround by having a separate `Invoke` impl that execs the istioctl binary. Downside is things are harder to debug (we can control the impl with a flag?)

**Comment by howardjohn** [comment]
> It's not.. but this part also doesn't seem to be horribly affected.
> 
> We could also workaround by having a separate Invoke impl that execs the istioctl binary. Downside is things are harder to debug (we can control the impl with a flag?)

It also makes it harder to run since the `go test` is no longer hermetic - depends on istioctl being built correctly, in the path, and not the wrong version, etc. Would prefer to avoid if possible

**Comment by nmittler** [comment]
@howardjohn 

> It also makes it harder to run since the go test is no longer hermetic - depends on istioctl being built correctly, in the path, and not the wrong version, etc. Would prefer to avoid if possible

+1 ... we specifically moved to the in-memory model because calling out to the command line for istioctl was incredibly confusing and error-prone when running locally.

**Comment by stevenctl** [comment]
My vote is to turn off the locking mechanism for install only, at least until we determine things area breaking there as well.

**Comment by nmittler** [comment]
@stevenctl 

> My vote is to turn off the locking mechanism for install only, at least until we determine things area breaking there as well.

I've already changed the installation code (#37332) to avoid using `istioctl.Invoke`, so this *should* be safe as-is.

**Comment by stevenctl** [comment]
This lgtm then.

**Review by stevenctl — APPROVED** [review]

**Comment by nmittler** [comment]
@stevenctl FYI with https://github.com/istio/istio/pull/37332 in place, this change seems to have virtually no impact on job run duration

**Pull request merged**